### ISSUE TITLE
[Bug] I can no longer share a link from another app directly to another device without opening it on my phone first.

### ISSUE THREAD

**Issue by fvbommel** [mozilla-mobile/fenix#14216]
## Steps to reproduce
1. Open an app that allows using Android's built-in "share" functionality (e.g. the Google app)
2. Share a link to Firefox
3. Send that link to another device

### Expected behavior
I used to be able to send a link to another device without first opening it on my phone. This should be the bare minimum IMHO: please don't remove useful functionality.

Alternatively (and preferably): instead of sharing links to "Firefox", allow me to share directly to "Firefox on \<other device\>" without first opening it on my phone. This should be possible, because other apps already allow this (e.g. WhatsApp allowing me to share to recent conversations).

### Actual behavior
After step 2 it opens the page in the mobile browser and I need two extra taps (open menu, share) before I can select a device to share to.

### Additional info
This used to work before a recent update of Firefox for Android (the one that changed the UI and put the address bar at the bottom by default).

Because the new workflow forces me to open it on my phone first, it also forces it to use data on my device (which when I'm not on WiFi means using my limited mobile data bundle) to load something I didn't want to open on my phone in the first place.

I often send stuff to my laptop as a way to "look at this later", especially for data-heavy sites or sites that are hard or unpleasant to use on small screens.

### Device information

* Android device: Motorola Moto G5 Plus (XT1685), Android 8.1.0
* Fenix version: Firefox 79.0.5 (Build #2015758617)

**Comment by Dunexus** [comment]
Duplicate of https://github.com/mozilla-mobile/fenix/issues/1523
